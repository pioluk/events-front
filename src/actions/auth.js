/* @flow */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from '../constants/auth'

export function loginRequest (username: string, password: string) {
  return {
    type: LOGIN_REQUEST,
    username,
    password
  }
}

export function loginSuccess (user: string, token: string) {
  return {
    type: LOGIN_SUCCESS,
    user,
    token
  }
}

export function loginFailure (error: Error) {
  return {
    type: LOGIN_FAILURE,
    error
  }
}

export function logout () {
  return {
    type: LOGOUT
  }
}
