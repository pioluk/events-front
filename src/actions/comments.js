// @flow

import {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_FAILURE,
  ADD_COMMENT,
  ADD_COMMENT_SUCCESS,
  ADD_COMMENT_FAILURE,
  REMOVE_COMMENT,
  REMOVE_COMMENT_SUCCESS,
  REMOVE_COMMENT_FAILURE,
} from '../constants/comments'

export function fetchComments (eventId: number, limit: number, offset: number) {
  return {
    type: FETCH_COMMENTS,
    eventId,
    limit,
    offset
  }
}

export function fetchCommentsSuccess (count: number, comments: any[]) {
  return {
    type: FETCH_COMMENTS_SUCCESS,
    count,
    comments
  }
}

export function fetchCommentsFailure (error: Error) {
  return {
    type: FETCH_COMMENTS_FAILURE,
    error
  }
}

export function addComment (text: string) {
  return {
    type: ADD_COMMENT,
    text
  }
}

export function addCommentSuccess (comment: any) {
  return {
    type: ADD_COMMENT_SUCCESS,
    comment
  }
}

export function addCommentFailure (error: Error) {
  return {
    type: ADD_COMMENT_FAILURE,
    error
  }
}

export function removeComment (id: number) {
  return {
    type: REMOVE_COMMENT,
    id
  }
}

export function removeCommentSuccess (id: number) {
  return {
    type: REMOVE_COMMENT_SUCCESS,
    id
  }
}

export function removeCommentFailure (error: Error) {
  return {
    type: REMOVE_COMMENT_FAILURE,
    error
  }
}
