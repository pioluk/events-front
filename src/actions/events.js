import {
  FETCH_EVENTS_REQUEST,
  FETCH_EVENTS_SUCCESS,
  FETCH_EVENTS_FAILURE
} from '../constants/events'

export function fetchEventsRequest () {
  return {
    type: FETCH_EVENTS_REQUEST
  }
}

export function fetchEventsSuccess (events) {
  return {
    type: FETCH_EVENTS_SUCCESS,
    events
  }
}

export function fetchEventsFailure (error) {
  return {
    type: FETCH_EVENTS_FAILURE,
    error
  }
}
