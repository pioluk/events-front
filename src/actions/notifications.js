import { DISPLAY_NOTIFICATION } from '../constants/notifications'

export function displayNotification (message) {
  return {
    type: DISPLAY_NOTIFICATION,
    message
  }
}
