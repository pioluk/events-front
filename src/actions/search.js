// @flow

import {
  SEARCH_EVENTS,
  SEARCH_EVENTS_RESET,
  SEARCH_EVENTS_SUCCESS
} from '../constants/search'

export function searchEvents (page: number, query: string) {
  return {
    type: SEARCH_EVENTS,
    page,
    query
  }
}

export function searchEventsReset () {
  return {
    type: SEARCH_EVENTS_RESET
  }
}

export function searchEventsSuccess (count: number, events: Array<any>) {
  return {
    type: SEARCH_EVENTS_SUCCESS,
    count,
    events
  }
}
