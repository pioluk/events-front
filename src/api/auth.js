/* @flow */

import { apiPost } from './utils'

export const login = (username: string, password: string) =>
  apiPost('login', { username, password })
