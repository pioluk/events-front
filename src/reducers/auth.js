/* @flow */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from '../constants/auth'

type AuthState = {
  errorMessage: ?string,
  isAuthenticated: boolean,
  isAuthenticating: boolean,
  token: ?string,
  user: ?any
}

const initialState: AuthState = {
  errorMessage: null,
  isAuthenticated: false,
  isAuthenticating: false,
  token: null,
  user: null
}

export default function authReducer (state: AuthState = initialState, action: any) {
  switch (action.type) {
    case LOGIN_REQUEST:
      return Object.assign({}, state, { isAuthenticating: true })

    default:
      return state
  }
}
