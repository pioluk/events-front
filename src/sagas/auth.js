import { call, take, put } from 'redux-saga/effects'
import { browserHistory } from 'react-router'
import { LOGIN_REQUEST } from '../constants/auth'
import * as authActions from '../actions/auth'
import * as notificationActions from '../actions/notifications'
import { login } from '../api/auth'

export function* loginFlow () {
  while (true) {
    const { username, password } = yield take(LOGIN_REQUEST);
    try {
      const { user, token } = yield call(login, username, password);
      if (user && token) {
        localStorage.setItem('user', JSON.stringify(user));
        localStorage.setItem('token', token);
        yield put(authActions.loginSuccess(user, token));
        yield put(notificationActions.displayNotification('Successfully logged in.'))
        browserHistory.push('/');
      }
    }
    catch (err) {
      yield put(authActions.loginFailure(err))
      yield put(notificationActions.displayNotification('Error: ' + err.message))
    }
  }
}
