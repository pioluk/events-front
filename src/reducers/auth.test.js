// global describe,it,expect

import reducer from './auth'
import * as actions from '../actions/auth'

describe('auth reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle LOGIN_REQUEST', () => {
    const username = 'admin'
    const password = 'admin1'
    const state = reducer(undefined, actions.loginRequest(username, password))
    expect(state).toMatchSnapshot()
  })

  it('should handle LOGIN_SUCCESS', () => {
    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ'
    const user = 'admin'
    const state = reducer(undefined, actions.loginSuccess(user, token))
    expect(state).toMatchSnapshot()
  })

  it('should handle LOGIN_FAILURE', () => {
    const error = new Error('Login failure.')
    const state = reducer(undefined, actions.loginFailure(error))
    expect(state).toMatchSnapshot()
  })

  it('should handle LOGOUT', () => {
    expect(reducer(undefined, actions.logout())).toMatchSnapshot()
  })
})
