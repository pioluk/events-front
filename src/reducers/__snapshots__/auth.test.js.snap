exports[`auth reducer should handle LOGIN_FAILURE 1`] = `
Object {
  "errorMessage": "Login failure.",
  "isAuthenticated": false,
  "isAuthenticating": false,
  "token": null,
  "user": null,
}
`;

exports[`auth reducer should handle LOGIN_REQUEST 1`] = `
Object {
  "errorMessage": null,
  "isAuthenticated": false,
  "isAuthenticating": true,
  "token": null,
  "user": null,
}
`;

exports[`auth reducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "errorMessage": null,
  "isAuthenticated": true,
  "isAuthenticating": false,
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ",
  "user": "admin",
}
`;

exports[`auth reducer should handle LOGOUT 1`] = `
Object {
  "errorMessage": null,
  "isAuthenticated": false,
  "isAuthenticating": false,
  "token": null,
  "user": null,
}
`;

exports[`auth reducer should return the initial state 1`] = `
Object {
  "errorMessage": null,
  "isAuthenticated": false,
  "isAuthenticating": false,
  "token": null,
  "user": null,
}
`;
